app/views/helpers/PageWithNavigation.scala

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views.helpers
+
+object PageWithNavigation extends Enumeration {
```

Comment:
This construct confuses me. It seems to be serving two different purposes, enabling and disabling the navigation links and optionally specifying what page we are on (dashboard or inbox). 

Expressions like `PageWithNavigation.OnPageWithoutNavigation` read as a contradiction of terms.

Would not a simpler solution be that each view that requires the navigation links pass them as an optional component to the gdsMainTemplate (disabling whichever link might need disabling)?

Reply:
Not sure if it helps, but I have an idea to instead of having this Enumeration object, change `gdsMainTemplate` parameter to be the an optional navigation banner itself instead of a flag.
This way you could have default of `None` and for pages where you need the navigation banner you could provide it as any other parameter. 

I'm happy to talk about it if it is not explained well enough ;-)

Reply:
And now I realized that I might be talking about the same thing as @glidester

Reply:
Its just optionally disabling one of the two links isn't it? The same functionality could be achieved with just two Boolean parameters on 'navigationBanner' where the loading view could disable either link as appropriate. Alternatively, if 'PageWithNavigation' just had two values of 'DashboardPage' and 'MessageInboxPage' then that would be clearer.

Reply:
Sorry, Friday I was in a hurry, so I didn't make myself clear.

The banner has to be always shown on four pages (dashboard, decl information, inbox, single conversation) but only if the Secure Messaging flag is enabled. What is different between the four pages is the style to be used
- Dashboard and inbox have one link only, the other, for the page itself, become just a span tag.
- "decl information" and "single conversation" have instead two links.

While your solution, passing to gdsMainTemplate what every page needs, is simpler also a few cons
- If not there yet you need to inject SecureMessagingConfig to every page.
- On the other side, if you do inject SecureMessagingConfig only in gdsMainTemplate you will end up creating the html in every page but they won't be shown if the flag is disabled.
- If in future you need to change the href url for one of the two links you have to change it in all pages.
- If in future you need to add a link to the banner you have to add it to all pages.

I have removed now the confusing enum item and setting the banner as optional in gdsMainTemplate. This solution still seems quite clean, to me at least.